web-ng/src/app/components/map/map.component.ts

```diff
@@ -165,11 +175,21 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     });
   }
 
+  private removePolygonOnMap(newFeatureIds: List<string>) {
+    this.map.data.forEach(f => {
```

Comment:
I don't think you should use `this.map.data` here, unlike geojson, the polygons are not added to the `this.map.data`. They are more like the Marker objects, they are added to/removed from the map using `.setMap(this.map)` or `.setMap(null)`.

You can try adding/removing polygons in firebase and see if it reflects on the map.

Reply:
updated.